RFS/src/main/java/org/opensearch/migrations/bulkload/delta/DeltaDocumentReaderEngine.java

```diff
@@ -53,24 +58,65 @@ public SnapshotShardUnpacker createUnpacker(
         );
     }
 
+    @SneakyThrows
```

Comment:
Please don't use sneaky throws lightly, lets capture and log the exception if there is a problem closing.